staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -306,6 +308,26 @@ func NoNamespaceKeyFunc(ctx context.Context, prefix string, name string) (string
 	return key, nil
 }
 
+// NoNamespaceReverseKeyFunc is the default function for constructing storage paths to
+// a resource relative to the given prefix enforcing namespace rules. If the
+// context does not contain a namespace, it errors.
+func NoNamespaceReverseKeyFunc(key string) (name string, namespace string, err error) {
+	tokens := strings.Split(key, "/")
+	return tokens[len(tokens)-1], "", nil
+}
+
+// NamespaceReverseKeyFunc is the default function for constructing storage paths to
+// a resource relative to the given prefix enforcing namespace rules. If the
+// context does not contain a namespace, it errors.
+func NamespaceReverseKeyFunc(key string) (name string, namespace string, err error) {
+	tokens := strings.Split(key, "/")
+	if len(tokens) < 2 {
```

Comment:
What happens if `len(tokens) > 2` we should return error instead of returning invalid response.